src/reconcile/rec_visibility.c

```diff
@@ -252,6 +257,112 @@ __timestamp_out_of_order_fix(WT_SESSION_IMPL *session, WT_TIME_WINDOW *select_tw
     return (false);
 }
 
+/*
+ * __rec_validate_upd_chain --
+ *     Check the update chain for conditions that would prevent its insertion into the history
+ *     store. Return EBUSY if the update chain cannot be inserted into the history store at this
+ *     time.
+ */
+static int
+__rec_validate_upd_chain(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *select_upd,
+  WT_TIME_WINDOW *select_tw, WT_CELL_UNPACK_KV *vpack)
+{
+    WT_DECL_RET;
+    WT_UPDATE *prev_upd, *upd;
+
+    /*
+     * There is no selected update to go to disk as such we don't need to check the updates
+     * following it.
+     */
+    if (select_upd == NULL)
+        return (0);
+
+    /* There is no history store so it is safe to reconcile out of order updates. */
```

Comment:
What is the case where there isn't a history store? Is it during startup or recovery or something else? Those seem like unusual conditions for this code to handle, so I think it's worth expanding the comment, or adding an assertion that the system is in the expected phase.

Reply:
When using in memory data base there is no history store. I also want to exclude fixed length column store but you just reminded me that if we have a fixed length column store we may still have a history store. So I need to modify this check.